galaxy_ng/app/api/v3/serializers/collection.py

```diff
@@ -26,10 +27,12 @@ def _get_href(self, url_name, **kwargs):
 
         view_namespace = self.context["view_namespace"]
 
-        # Handle the case where the /api/automation-hub/v3/collections/  urls do not
+        # Handle the case where the /api/automation-hub/v3/collections/ urls do not
         # include a "<str:path>" path param. ie, the bacwares compatible default "golden" repo
-        if "/<str:path>/v3/" not in self.context["view_route"]:
-            del kwargs["path"]
+        if "/<str:path>/v3/" in self.context["view_route"]:
+            kwargs.setdefault("path", self.context["path"])
+        else:
+            kwargs.pop("path", None)
```

Comment:
Unrelated to this pr and not something that needs to be addressed now, but eventually I'd like to change 'path' to something more specific  (ie, 'distro_base_path' or slightly inaccurately 'repo_name' or 'repo_path')